packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -348,8 +346,18 @@ class TreeViewExtImpl<T> implements Disposable {
         return idLabel;
     }
 
-    private buildTreeItemId(parentId: string, index: number, idLabel: string | undefined): string {
-        return `${parentId}/${index}:${idLabel}`;
+    private buildTreeItemId(parentId: string, item: TreeItem): string {
+        // build tree id according to https://code.visualstudio.com/api/references/vscode-api#TreeItem
+        // note: the front end tree implementation cannot handle reparenting items, hence the id is set to the "path" of individual ids
+        let id = typeof item.id === 'string' ? item.id : this.getItemLabel(item);
+        if (id) {
+            // we use '' as the id of the root, we don't consider that a valid id
+            // since '' is falsy, we'll never get '' in this branch
+            id = TreeViewExtImpl.ID_ITEM + id;
+        } else {
+            id = TreeViewExtImpl.ID_COMPUTED + this.nextItemId++;
+        }
+        return `${parentId}/${id}`;
```

Comment:
Since we're always adding additional material, wether we find an existing ID or not, if we have a long-lived tree, will it gradually accrete infinitely long ID's, or is there some guarantee that we'll start fresh at some point?

Reply:
A 64 bit number maxes out at around 20 digits, so the individual id will be max 21 chars. Since we're using paths, the average path length will by something like 21 * log(n) (n= number of items). With 100000 items _visible_ in the tree, we get 1000000*5*21 which is somewhere around 10mb. Should be fine.

Note that the id here is not the id of of the TreeItem, so the id's don't get longer.

Reply:
Ah - the initial ID is of the API item, not a known TreeNode.